src/targets/gpu/kernels/include/migraphx/kernels/type_traits.hpp

```diff
@@ -218,19 +218,23 @@ using common_type_t = typename common_type<Ts...>::type;
 
 #define MIGRAPHX_REQUIRES(...) class = enable_if_t<__VA_ARGS__>
 
-constexpr unsigned long int_max(unsigned long n) { return (1u << (n * 8)) - 1; }
+// Note, this cannot be used to get the maximum value of int64_type or
+// uint64_type because it is undefined behavior to left shift 64 bits for
+// these types
+constexpr unsigned long int_max(unsigned long n) { return (1ul << (n * 8)) - 1; }
```

Comment:
Maybe we could do `if (n == 8) return -1;` to handle the 64 bit case?

Reply:
That sounds to be a good idea!

Reply:
Done in dc852557c2ab6cb6c44e4ac140225c1413906a26